src/components/navigation/context/getNode.ts

```diff
@@ -1,17 +1,24 @@
 import { TreeNode } from '../types';
+import isComponentNode from '../utils';
+
+type ReturnType = TreeNode | null;
 
 const getNode = (
   tree: TreeNode[] | TreeNode,
   key: string,
   value?: string,
-): TreeNode | null => {
+): ReturnType => {
   if (Array.isArray(tree)) {
-    return tree.reduce(
-      (acc: null | TreeNode, node) => acc || getNode(node, key, value),
+    return tree.reduce<ReturnType>(
+      (acc, node) => acc || getNode(node, key, value),
       null,
     );
   }
 
+  if (isComponentNode(tree)) {
+    return null;
```

Comment:
We return null here, as we don't really need to have those events for the custom components. And we have no good way of doing that anyways too.
Yes, not ideal. But what's ideal in this life?

Reply:
I was wrong on this, had to redo this